UnityProject/Assets/Scripts/Player/PlayerAction.cs

```diff
@@ -22,14 +31,55 @@ public Vector2Int Direction()
 		Vector2Int direction = Vector2Int.zero;
 		for (var i = 0; i < moveActions.Length; i++)
 		{
-			direction += GetMoveDirection((MoveAction)moveActions[i]);
+			direction += GetMoveDirection((MoveAction) moveActions[i]);
 		}
+
 		direction.x = Mathf.Clamp(direction.x, -1, 1);
 		direction.y = Mathf.Clamp(direction.y, -1, 1);
 
 		return direction;
 	}
 
+	public MovementSynchronisation.PlayerMoveDirection ToPlayerMoveDirection()
+	{
+		var direction = Direction();
+
+		if (direction == Vector2Int.up)
+		{
+			return MovementSynchronisation.PlayerMoveDirection.Up;
+		}
+		else if (direction == Vector2Int.down)
+		{
+			return MovementSynchronisation.PlayerMoveDirection.Down;
+		}
+		else if (direction == Vector2Int.left)
+		{
+			return MovementSynchronisation.PlayerMoveDirection.Left;
+		}
+		else if (direction == Vector2Int.right)
+		{
+			return MovementSynchronisation.PlayerMoveDirection.Right;
+		}
+		else if (direction == Up_Right)
+		{
+			return MovementSynchronisation.PlayerMoveDirection.Up_Right;
+		}
+		else if (direction == Down_Right)
+		{
+			return MovementSynchronisation.PlayerMoveDirection.Down_Right;
+		}
+		else if (direction == Left_Down)
+		{
+			return MovementSynchronisation.PlayerMoveDirection.Down_Left;
+		}
+		else if (direction == Up_Left)
+		{
+			return MovementSynchronisation.PlayerMoveDirection.Up_Right;
+		}
+
+		return MovementSynchronisation.PlayerMoveDirection.Up;
```

Comment:
Probably better to use a static dictionary here.

Reply:
ok will do